src/parameterized_tendencies/les_sgs_models/smagorinsky_lilly.jl

```diff
@@ -107,22 +107,27 @@ function horizontal_smagorinsky_lilly_tendency!(Yₜ, Y, p, t, ::SmagorinskyLill
     @. Yₜ.c.ρe_tot += wdivₕ(Y.c.ρ * ᶜD_smag * gradₕ(ᶜh_tot))
 
     ## Tracer diffusion and associated mass changes
-    for (ᶜρχₜ, ᶜχ, χ_name) in CA.matching_subfields(Yₜ.c, ᶜspecific)
-        χ_name == :e_tot && continue
-        ᶜρχₜ_diffusion =
-            @. p.scratch.ᶜtemp_scalar = wdivₕ(Y.c.ρ * ᶜD_smag * gradₕ(ᶜχ))
-        @. ᶜρχₜ += ᶜρχₜ_diffusion
-        # Rain and snow does not affect the mass
-        if χ_name ∉ (:q_rai, :q_sno)
-            @. Yₜ.c.ρ += ᶜρχₜ_diffusion
+    for ρχ_name in propertynames(Yₜ.c)
+        if is_tracer_var(ρχ_name) && ρχ_name != :ρe_tot
+            ᶜρχ = getproperty(Y.c, ρχ_name)
+            ᶜρχₜ = getproperty(Yₜ.c, ρχ_name)
+            ᶜχ = @. lazy(specific(ᶜρχ, Y.c.ρ))
+            ᶜρχₜ_diffusion = @. lazy(wdivₕ(Y.c.ρ * ᶜD_smag * gradₕ(ᶜχ)))
```

Comment:
I think this is better as (to avoid nested lazy's)
```suggestion
            ᶜρχₜ_diffusion = @. lazy(wdivₕ(Y.c.ρ * ᶜD_smag * gradₕ(specific(ᶜρχ, Y.c.ρ))))
```

Reply:
^^^ @dennisYatunin. My understanding is that they can generate compilation problems if the nesting is too deep.

Reply:
I'll inline it.